client/test/backend.go

```diff
@@ -160,12 +173,99 @@ func outcomeRecursive(state *channel.State, subStates channel.StateMap) (outcome
 
 // Withdraw withdraws the channel funds.
 func (b *MockBackend) Withdraw(_ context.Context, req channel.AdjudicatorReq, subStates channel.StateMap) error {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+
+	// Check concluded.
+	ch := req.Params.ID()
+	if b.isConcluded(ch) {
+		log.Debug("withdraw: already concluded:", ch)
+		return nil
+	}
+
 	outcome := outcomeRecursive(req.Tx.State, subStates)
 	b.log.Infof("Withdraw: %+v, %+v, %+v", req, subStates, outcome)
+	for a, assetOutcome := range outcome {
+		asset := req.Tx.Allocation.Assets[a]
+		for p, amount := range assetOutcome {
+			participant := req.Params.Parts[p]
+			b.addBalance(participant, asset, amount)
```

Comment:
I dont quite get how it workks.  
You redistribute the on-balances here, but the `Fund` function never debited the accounts?  
Where the balance subtracted from the accounts?

Reply:
We discussed that it only tests the optimistic case and therefore assumed correct funding.